src/select/SelectOption.ts

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import {
+  useCompositeItem,
+  CompositeItemHTMLProps,
+  CompositeItemOptions,
+} from "reakit/Composite";
+import { SELECT_KEYS } from "./__keys";
+import { SelectStateReturn } from "./SelectState";
+import { createHook, createComponent } from "reakit-system";
+import { useLiveRef, hasFocusWithin } from "reakit-utils";
+
+export type SelectItemOptions = CompositeItemOptions &
+  Pick<SelectStateReturn, "setSelected" | "selected"> & {
+    value: string;
+  };
+
+export type SelectItemHTMLProp = CompositeItemHTMLProps;
+
+const useSelectOption = createHook<SelectItemOptions, SelectItemHTMLProp>({
+  name: "SelectOption",
+  compose: useCompositeItem,
+  keys: ["value", ...SELECT_KEYS],
+  useProps(
+    { selected, value, setSelected, disabled },
+    { ref: htmlRef, onMouseMove: htmlOnMouseMove, ...htmlProps },
```

Comment:
No need to extract ref anymore

```suggestion
    {  onMouseMove: htmlOnMouseMove, ...htmlProps },
```